astropy/io/asdf/extension.py

```diff
@@ -0,0 +1,41 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+import os
+
+# Make sure that all tag implementations are imported by the time we create
```

Comment:
Is this something that can be changed later? I'd assume it would be done in `tags.__init__` - is it not possible to have a `try/except` around an `asdf` import there?

I guess I'm mostly asking to clarify the comment here even more, including what the future final state should be.

Reply:
It might be possible to do this with `try/except` but I think it would have to be done with every `__init__.py` in the `astropy/io/asdf` tree in order to avoid import errors at each level. I'm not sure if that would be any cleaner, but I'm open to suggestions.